changelogs/unreleased/gh-10715-memtx-mvcc-conflict-with-prepared-sysview-txn.md

```diff
@@ -0,0 +1,5 @@
+## bugfix/memtx
```

Comment:
> To solve the problem, we could try to write quite a large patch to handle
such transactions differently in memtx MVCC. But instead, let's simply
finalize prepare of all transaction in MVCC. The current design already
lacks encapsulation - all transactions (even non-memtx) are registered
there, so such solution not only simple but also consistent with other
MVCC parts. Note that memtx MVCC still can abort such non-memtx
transactions when they are in-progress, but it seems that such behavior
doesn't break other yielding engines (`vinyl` is the only one).

I really dislike the motivation for this solution, it's totally a crudge. What if tomorrow the sysview engine (or one of the other transaction management bypassing engines) starts using other engines' metadata?

What's wrong with the following solutions?
1. Add a new helper around the `memtx_use_mvcc_engine` flag that will also check for the  `ENGINE_BYPASS_TX` flag set on a transaction (or check that transaction actually uses the memtx engine).
2. Rework the original `ENGINE_BYPASS_TX` flag solution: AFAIC, the engines that use it are actually read-only, so mixing them with other engines shouldn't break anything. We could introduce some `ENGINE_READ_ONLY` flag and avoid raising `ER_CROSS_ENGINE_TRANSACTION` if the transaction uses only read-only engines.